### PR TITLE
InstancedMesh: Add support for morph targets.

### PR DESCRIPTION
**Pull request by wizgrav**
@Mugen87 I took a stub at adding full morph target support as I told you on the other PR and the results were surprisingly good. I don't notice any performance difference with the horse model which has 15 targets so it seems quite performant. 

Now everything is loaded on a texture including the base influence so it will support every case and the API is identical to the attribute approach. I left the example as is and will make some more to test other cases but wont push in this PR

[demo here](https://wizgrav.github.io/three.js/examples/?q=morph#webgl_instancing_morph) (you may need to refresh cache to be sure, output is identical to the previous one)

### CODE REVIEW

**Review by Mugen87 — APPROVED** [review]
The solution looks awesome now!

**Comment by wizgrav** [comment]
@Mugen87 I wrote an example using the face and it's slower than the horse but that was not unexpected. I havent pushed it in the PR or even added it in the examples list but you can [check it out directly here](https://wizgrav.github.io/three.js/examples/webgl_morphtargets_faces.html)

I'll do a quick test to see if using a half float texture makes any difference

File: src/renderers/shaders/ShaderLib/linedashed.glsl.js [hunk 1]
```diff
@@ -18,6 +18,7 @@ void main() {
 
 	#include <uv_vertex>
 	#include <color_vertex>
+	#include <morphtarget_instancing>
```
[the remark this answers is not in the document]
Comment (wizgrav) [reply src/renderers/shaders/ShaderLib/linedashed.glsl.js]: sure done

**Comment by Mugen87** [comment]
Regarding the code complexity in `WebGLRenderer`: Yes, this PR adds additional complexity to the renderer but also adds a useful feature. When WebGL 1 support is removed next month, the renderer can be simplified at certain places. E.g. the entire WebGL 1 code path for morph targets can be deleted. That should keep the morph target related code manageable even with this PR in place.

**Pull request merged**